Compilation error in MacOS with llvm17

```
dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp:283:21: error: call to constructor of 'RegionDataMemDiff' is ambiguous
  283 |     recordMemChange(RegionDataMemDiff{-cf_data_size, -decoded_data_size});
      |                     ^                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:32:5: note: candidate constructor
   32 |     RegionDataMemDiff(Type payload_, Type decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:36:5: note: candidate constructor
   36 |     RegionDataMemDiff(UInt64 payload_, UInt64 decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp:291:23: error: call to constructor of 'RegionDataMemDiff' is ambiguous
  291 |     updateMemoryUsage(RegionDataMemDiff{rhs.cf_data_size, rhs.decoded_data_size});
      |                       ^                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:32:5: note: candidate constructor
   32 |     RegionDataMemDiff(Type payload_, Type decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:36:5: note: candidate constructor
   36 |     RegionDataMemDiff(UInt64 payload_, UInt64 decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp:359:21: error: call to constructor of 'RegionDataMemDiff' is ambiguous
  359 |     recordMemChange(RegionDataMemDiff{-cf_data_size, 0});
      |                     ^                ~~~~~~~~~~~~~~~~~~
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:32:5: note: candidate constructor
   32 |     RegionDataMemDiff(Type payload_, Type decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:36:5: note: candidate constructor
   36 |     RegionDataMemDiff(UInt64 payload_, UInt64 decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp:360:23: error: call to constructor of 'RegionDataMemDiff' is ambiguous
  360 |     updateMemoryUsage(RegionDataMemDiff{-cf_data_size, 0});
      |                       ^                ~~~~~~~~~~~~~~~~~~
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:32:5: note: candidate constructor
   32 |     RegionDataMemDiff(Type payload_, Type decoded_)
      |     ^
dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h:36:5: note: candidate constructor
   36 |     RegionDataMemDiff(UInt64 payload_, UInt64 decoded_)
      |     ^
4 errors generated.
[155/1049] Building CXX object dbms/src/Storages/KVStore/CMakeFiles/kvstore.dir/MultiRaft/Disagg/CheckpointIngestInfo.cpp.o
ninja: build stopped: subcommand failed.

```


https://github.com/pingcap/tiflash/blob/d27d7c0b741369f1016e64415b1be844bcc11ec6/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h#L26-L43

These two function overloads are too similar...

--- a/dbms/src/Storages/KVStore/MultiRaft/Disagg/RaftLogManager.h
+++ b/dbms/src/Storages/KVStore/MultiRaft/Disagg/RaftLogManager.h
@@ -18,6 +18,7 @@
 #include <Storages/KVStore/Types.h>
 #include <common/types.h>
 
+#include <mutex>
 #include <unordered_map>
 
 namespace DB

--- a/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
+++ b/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
@@ -29,7 +29,7 @@ struct RegionDataMemDiff
     Type payload;
     Type decoded;
 
-    template<typename T1, typename T2>
+    template <typename T1, typename T2>
     RegionDataMemDiff(T1 payload_, T2 decoded_)
         : payload(payload_)
         , decoded(decoded_)

--- a/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
+++ b/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
@@ -29,14 +29,12 @@ struct RegionDataMemDiff
     Type payload;
     Type decoded;
 
-    RegionDataMemDiff(Type payload_, Type decoded_)
+    template<typename T1, typename T2>
+    RegionDataMemDiff(T1 payload_, T2 decoded_)
         : payload(payload_)
         , decoded(decoded_)
     {}
-    RegionDataMemDiff(UInt64 payload_, UInt64 decoded_)
-        : payload(static_cast<Type>(payload_))
-        , decoded(static_cast<Type>(decoded_))
-    {}
+
     RegionDataMemDiff()
         : payload(0)
         , decoded(0)

--- a/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
+++ b/dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h
@@ -29,8 +29,7 @@ struct RegionDataMemDiff
     Type payload;
     Type decoded;
 
-    template <typename T1, typename T2>
-    RegionDataMemDiff(T1 payload_, T2 decoded_)
+    RegionDataMemDiff(Type payload_, Type decoded_)
         : payload(payload_)
         , decoded(decoded_)
     {}

--- a/dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp
+++ b/dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp
@@ -280,7 +280,7 @@ size_t RegionData::totalSize() const
 
 void RegionData::assignRegionData(RegionData && rhs)
 {
-    recordMemChange(RegionDataMemDiff{-cf_data_size, -decoded_data_size});
+    recordMemChange(RegionDataMemDiff{-cf_data_size.load(), -decoded_data_size.load()});
     resetMemoryUsage();
 
     default_cf = std::move(rhs.default_cf);
@@ -288,7 +288,7 @@ void RegionData::assignRegionData(RegionData && rhs)
     lock_cf = std::move(rhs.lock_cf);
     orphan_keys_info = std::move(rhs.orphan_keys_info);
 
-    updateMemoryUsage(RegionDataMemDiff{rhs.cf_data_size, rhs.decoded_data_size});
+    updateMemoryUsage(RegionDataMemDiff{rhs.cf_data_size.load(), rhs.decoded_data_size.load()});
     rhs.resetMemoryUsage();
 }
 

--- a/dbms/src/Storages/KVStore/MultiRaft/RegionData.h
+++ b/dbms/src/Storages/KVStore/MultiRaft/RegionData.h
@@ -135,9 +135,9 @@ private:
     OrphanKeysInfo orphan_keys_info;
 
     // Size of 3 cfs, reflects size of real payload flows to KVStore.
-    std::atomic<size_t> cf_data_size = 0;
+    std::atomic<Int64> cf_data_size = 0;
     // Size of decoded structures for convenient access, considered as amplification in memory.
-    std::atomic<size_t> decoded_data_size = 0;
+    std::atomic<Int64> decoded_data_size = 0;
 };
 
 } // namespace DB

--- a/libs/libcommon/include/common/demangle.h
+++ b/libs/libcommon/include/common/demangle.h
@@ -14,9 +14,9 @@
 
 #pragma once
 
+#include <cstdlib>
 #include <memory>
 #include <string>
-#include <cstdlib>
 
 
 /** Demangles C++ symbol name.

--- a/libs/libcommon/include/common/demangle.h
+++ b/libs/libcommon/include/common/demangle.h
@@ -16,6 +16,7 @@
 
 #include <memory>
 #include <string>
+#include <cstdlib>
 
 
 /** Demangles C++ symbol name.
@@ -37,7 +38,7 @@ struct FreeingDeleter
     template <typename PointerType>
     void operator()(PointerType ptr)
     {
-        std::free(ptr);
+        std::free(ptr); // NOLINT(cppcoreguidelines-no-malloc)
     }
 };
 

